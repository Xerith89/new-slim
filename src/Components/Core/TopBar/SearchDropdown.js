import React, { Component } from 'react'
import { faSearch } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export default class SearchDropdown extends Component {

    constructor(props){
        super(props)

        this.state = {
            type: 'Spec',
            number: '',
        }

    }

    handleChange = (event) => {
        let name = event.target.name;
        let target = event.target.value;
        this.setState({
            [name]: target}
        );
    }

    handleSearch = (event) => {
        event.preventDefault();
        if (this.state.number !== "") {
            console.log("Make request to API");
        }
    }

    render() {
        return (
            <div data-testid="searchMenu">
                <div className="list-group dropdown-menu dropdown-menu-right mr-5 text-center">
                    <h5><FontAwesomeIcon icon={faSearch}/> Quick Search</h5>
                    <form className="m-2">
                        <div className="form-group" onChange={this.handleChange}>
                            <input type="radio" value="Spec" name="type" defaultChecked /> Specification 
                            <input type="radio" value="Claim" name="type" className="ml-2"/> Claim
                        </div>
                        <div className="form-group">
                            <label>Search Number</label>
                            <input type="text" data-testid="searchText" onChange={this.handleChange} value={this.state.number} name="number" className="form-control"/>
                        </div>
                        <button className="btn btn-block btn-primary" onClick={this.handleSearch} data-testid="searchButton">Search</button>
                    </form>
                    <div className="panel panel-default">
                    </div>
                    <button className="bg-transparent border-0 link">Advanced Search</button>
                </div>
            </div>
        )
    }
}
