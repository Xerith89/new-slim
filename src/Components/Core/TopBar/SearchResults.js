import React, { Component } from 'react'
import {
    Button,
    Modal,
    ModalHeader,
    ModalBody,
} from 'reactstrap';

export default class SearchResults extends Component {
    state = {
        modal:false,
        name: ''
    }

    toggle = () => {
        this.setState({
            modal: !this.state.modal
        })
    }

    render() {
        return (
            <div>
                <Button color="dark" className="btn-block" data-testid="searchButton" onClick={this.toggle}>Search</Button>

                <Modal data-testid="searchResults" isOpen={this.state.modal} toggle={this.toggle}>
                    <ModalHeader toggle={this.toggle}>Search Results</ModalHeader>
                    <ModalBody>
                    </ModalBody>
                </Modal>
            </div>
        )
    }
}
