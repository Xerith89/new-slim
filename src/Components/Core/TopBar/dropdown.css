
a {
    text-decoration: none !important;
    color: #000000;
}
a:hover {
    color: #000000;
}
.btn {
    outline: none;
    background-repeat: repeat-x;
}

.link:hover {
    text-decoration: underline;
}

.dropdown-item:hover {
    background-color: #ebdede;
   
}
