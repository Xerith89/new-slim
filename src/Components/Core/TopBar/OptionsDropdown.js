import React from 'react'
import '../../../../public/css/dropdown.css'
import { faCog } from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import PropTypes from 'prop-types'

export default function OptionsDropdown(props) {
    return (
        <div data-testid="optionsMenu">
            <ul className="list-group dropdown-menu dropdown-menu-right mr-2 p-2 text-center">
                <h5><FontAwesomeIcon icon={faCog}/> Options</h5>
                {props.links.map((element, i) => {
                    return (<React.Fragment key={i}>
                            <div className="dropdown-divider"></div>
                            <li className="dropdown-item"><a href={`/${element}`}>{element}</a></li>
                            </React.Fragment>)
                })}
            </ul>
        </div>
    )
}

OptionsDropdown.propTypes = {
    links: PropTypes.array
};
