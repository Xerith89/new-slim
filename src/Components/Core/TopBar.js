import React from 'react'
import {Component} from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCog, faSearch } from '@fortawesome/free-solid-svg-icons'
import OptionsDropdown from './OptionsDropdown'
import PropTypes from 'prop-types'
import SearchDropdown from './SearchDropdown'

export default class TopBar extends Component {
    
    constructor(props) {
        super(props)

        this.state = {
            optionsOpen:false,
            searchOpen:false,
        }
    }

    toggleOptionsDropDown = () => {
        this.setState({
            optionsOpen: !this.state.optionsOpen,
            searchOpen: false,
        })
    }

    toggleSearchDropDown = () => {
        this.setState({
            optionsOpen: false,
            searchOpen: !this.state.searchOpen,
        })
    }
    
    render() {

        const optionsIsOpen = this.state.optionsOpen;
        const searchIsOpen = this.state.searchOpen;
        let menu;
        let search;

        if (optionsIsOpen) {
            menu = <OptionsDropdown/>
        }

        if (searchIsOpen) {
           search = <SearchDropdown/>
        }

        return (
           <div data-testid="topbar">
            <nav className="navbar navbar-expand-lg navbar-dark bg-dark p-0">
                <h3 className="text-white ml-5 mb-0">New SLIM</h3>
                <div className="collapse navbar-collapse" id="navbarSupportedContent">
                    <ul className="navbar-nav ml-auto">
                        <li className="nav-item nav-link active">
                            <h3>Welcome, {this.props.user}</h3>
                        </li>
                        <li className="nav-item ">
                           <button className="nav-link bg-transparent border-0" data-testid="searchToggler" onClick={this.toggleSearchDropDown}><FontAwesomeIcon icon={faSearch} size="2x"  /></button> 
                        </li>
                        {search}
                        <div className="dropdown pr-5">
                            <li className="nav-item">
                                <button className="nav-link bg-transparent border-0" data-testid="optionsToggler" onClick={this.toggleOptionsDropDown} ><FontAwesomeIcon icon={faCog} size="2x" /></button>
                            </li>
                            {menu}
                        </div>
                    </ul>
                </div>
            </nav>
        </div>
        )
    }
}

TopBar.propTypes = {
    user: PropTypes.string
};
