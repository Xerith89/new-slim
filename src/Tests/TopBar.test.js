import React from 'react';
import TopBar from '../Components/Core/TopBar/TopBar';
import {render, cleanup, fireEvent} from '@testing-library/react';


afterEach(cleanup)

test('TopBar component renders to the DOM', () => {
    const {getByTestId} = render(<TopBar />);
    expect(getByTestId('topbar')).toMatchSnapshot();
});

test('Cog toggles settings menu', () => {
    const {queryByText, getByTestId} = render(<TopBar user="Alex"/>)
    expect(queryByText('Logout')).toBeFalsy();
    expect(queryByText('Settings')).toBeFalsy();
    fireEvent.click(getByTestId('optionsToggler'));
    expect(queryByText('Logout')).toBeTruthy();
    expect(queryByText('Settings')).toBeTruthy();
    fireEvent.click(getByTestId('optionsToggler'));
    expect(queryByText('Logout')).toBeFalsy();
    expect(queryByText('Settings')).toBeFalsy();
});

test('Magnifying glass toggles search menu', () => {
    const {queryByText, getByTestId} = render(<TopBar user="Alex"/>)
    expect(queryByText('Quick Search')).toBeFalsy();
    fireEvent.click(getByTestId('searchToggler'));
    expect(queryByText('Quick Search')).toBeTruthy();
    fireEvent.click(getByTestId('searchToggler'));
    expect(queryByText('Quick Search')).toBeFalsy();
});

test('Clicking on an option closes the other', () => {
    const {queryByText, getByTestId} = render(<TopBar user="Alex"/>)
    fireEvent.click(getByTestId('searchToggler'));
    fireEvent.click(getByTestId('optionsToggler'));
    expect(queryByText('Logout')).toBeTruthy();
    expect(queryByText('Settings')).toBeTruthy();
    expect(queryByText('Quick Search')).toBeFalsy();
    fireEvent.click(getByTestId('searchToggler'));
    expect(queryByText('Logout')).toBeFalsy();
    expect(queryByText('Settings')).toBeFalsy();
    expect(queryByText('Quick Search')).toBeTruthy();
});
