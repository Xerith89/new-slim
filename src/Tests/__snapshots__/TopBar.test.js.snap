// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar component renders to the DOM 1`] = `
<div
  data-testid="topbar"
>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-dark p-0"
  >
    <h3
      class="text-white ml-5 mb-0"
    >
      New SLIM
    </h3>
    <div
      class="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        class="navbar-nav ml-auto"
      >
        <li
          class="nav-item nav-link active"
        >
          <h3>
            Welcome, 
          </h3>
        </li>
        <li
          class="nav-item "
        >
          <button
            class="nav-link bg-transparent border-0"
            data-testid="searchToggler"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-search fa-w-16 fa-2x "
              data-icon="search"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                fill="currentColor"
              />
            </svg>
          </button>
        </li>
        <div
          class="dropdown pr-5"
        >
          <li
            class="nav-item"
          >
            <button
              class="nav-link bg-transparent border-0"
              data-testid="optionsToggler"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-cog fa-w-16 fa-2x "
                data-icon="cog"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M487.4 315.7l-42.6-24.6c4.3-23.2 4.3-47 0-70.2l42.6-24.6c4.9-2.8 7.1-8.6 5.5-14-11.1-35.6-30-67.8-54.7-94.6-3.8-4.1-10-5.1-14.8-2.3L380.8 110c-17.9-15.4-38.5-27.3-60.8-35.1V25.8c0-5.6-3.9-10.5-9.4-11.7-36.7-8.2-74.3-7.8-109.2 0-5.5 1.2-9.4 6.1-9.4 11.7V75c-22.2 7.9-42.8 19.8-60.8 35.1L88.7 85.5c-4.9-2.8-11-1.9-14.8 2.3-24.7 26.7-43.6 58.9-54.7 94.6-1.7 5.4.6 11.2 5.5 14L67.3 221c-4.3 23.2-4.3 47 0 70.2l-42.6 24.6c-4.9 2.8-7.1 8.6-5.5 14 11.1 35.6 30 67.8 54.7 94.6 3.8 4.1 10 5.1 14.8 2.3l42.6-24.6c17.9 15.4 38.5 27.3 60.8 35.1v49.2c0 5.6 3.9 10.5 9.4 11.7 36.7 8.2 74.3 7.8 109.2 0 5.5-1.2 9.4-6.1 9.4-11.7v-49.2c22.2-7.9 42.8-19.8 60.8-35.1l42.6 24.6c4.9 2.8 11 1.9 14.8-2.3 24.7-26.7 43.6-58.9 54.7-94.6 1.5-5.5-.7-11.3-5.6-14.1zM256 336c-44.1 0-80-35.9-80-80s35.9-80 80-80 80 35.9 80 80-35.9 80-80 80z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </li>
        </div>
      </ul>
    </div>
  </nav>
</div>
`;
